includes/adminpages.php

```diff
@@ -248,6 +248,90 @@ function pmpro_admin_bar_menu() {
 }
 add_action( 'admin_bar_menu', 'pmpro_admin_bar_menu', 1000);
 
+/**
+ * Add the "Admin Membership Access" menu to the admin bar.
+ */
+function pmpro_admin_membership_access_menu_bar() {
+	global $wp_admin_bar, $current_user;
+
+	// Only show when viewing the frontend of the site.
+	if ( is_admin() ) {
+		return;
+	}
+
+	/**
+	 * Filter to hide the "Admin Membership Access" menu in the admin bar.
+	 * @since TBD
+	 * @param bool $hide Whether to hide the "Admin Membership Access" menu in the admin bar. Default false.
+	 */
+	if ( apply_filters( 'pmpro_hide_admin_membership_access_toolbar', false ) ) {
+		return;
+	}
+
+	// View menu at all?
+	if ( ! current_user_can( 'pmpro_memberships_menu' ) || ! is_admin_bar_showing() ) {
```

Comment:
```suggestion
	if ( ! current_user_can( 'manage_options' ) || ! is_admin_bar_showing() ) {
```

Reply:
Only admins should be able to view this, and let's avoid updating user meta unless they can manage_options.